### ISSUE TITLE
error with AST.Build

### ISSUE THREAD

**Issue by francescoagati** [back2dos/tinkerbell#1]
```
#if macro 
import haxe.macro.Expr;
import tink.macro.tools.AST;
#end

class Main {

  @:macro static public function setter(e:Expr, field:String) {
      return AST.build( {
          var tmp = $e;
          function (v) {
              return $e.eval__field = v;
          }
      });
  }

  public static function main(){
    var o = { foo: 5 };
    var s = setter(o, 'foo');
  }

}
```

with this code i have this error:

tink/macro/tools/Printer.hx:57: characters 9-18 : This constructor is not part of the enum haxe.macro.ComplexType
tink/macro/tools/Printer.hx:159: characters 10-20 : This constructor is not part of the enum haxe.macro.ExprDef
tink/macro/tools/Printer.hx:138: lines 138-236 : String should be Void
tink/macro/tools/Printer.hx:51: characters 4-8 : String should be Void

Is for haxe 2.0.8 Optional arguments in type?

**Comment by back2dos** [comment]
Hi francesco, 

TOptional is not included ComplexType in haXe 2.08 yet, but it's [already in the docs](http://haxe.org/api/haxe/macro/complextype). I think it was mostly included for encoding optional type parameters in function signatures.

I'm afraid you will have to use a [nightly build](http://haxe.cmt.tc/) or build haXe yourself. Also, I think we can expect a haXe 2.09 some time soon :)

**Comment by francescoagati** [comment]
thank resolved :-)
